bitcoin/examples/create-p2wpkh-address.rs

```diff
@@ -0,0 +1,23 @@
+extern crate bitcoin;
+
+use bitcoin::{KnownHrp, Address, CompressedPublicKey, PrivateKey, Network};
+use bitcoin::secp256k1::{rand, Secp256k1};
+
+fn main() {
+    // Create a new Secp256k1 elliptic curve.
+    let s = Secp256k1::new();
+
+    // Generate a Secp256k1 secret key.
+    let (secret_key, _) = s.generate_keypair(&mut rand::thread_rng());
+
+    // Create a Bitcoin private key.
+    let private_key = PrivateKey::new(secret_key, Network::Bitcoin);
+
+    // Create a compressed public key derived from the private key.
+    let public_key = CompressedPublicKey::from_private_key(&s, private_key).unwrap();
```

Comment:
In 77d8b4793cab782dc2e47c867b2a3e29f08eb884:

...only to throw it away here. You could more directly convert the original public key to a `CompressedPublicKey`.

Alternately, you could take both strategies and assert that they're equal.

Reply:
Thanks.  I think the simpler case as an example would be fine.

Reply:
fixed